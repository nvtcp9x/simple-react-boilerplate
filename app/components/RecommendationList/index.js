// @flow

import * as React from 'react';
import styled from 'styled-components';

import { URL_IMG, IMG_SIZE_AVERAGE } from '../../../utils/constants'

const Wrapper = styled.div`
  display: flex;
  overflow: auto;
  justify-content: space-between;
`
const StyledRecommendation = styled.div`
  margin-right: 10px;
  margin-bottom: 7px;
`

const StyledImg = styled.img`
  border-radius: 5px;
`
const RecommendationList = ({ recommendations }: Array) => (
  <Wrapper>
    { recommendations.map(recommendation =>
      (
        <StyledRecommendation key={recommendation.id}>
          <StyledImg
            src={recommendation.backdrop_path ? `${URL_IMG}${IMG_SIZE_AVERAGE}${recommendation.backdrop_path}` : '/images/face.jpg'}
            alt={recommendation.title}
          />
          <p>{recommendation.title}</p>
        </StyledRecommendation>
      ))}
  </Wrapper>
)


export default RecommendationList
