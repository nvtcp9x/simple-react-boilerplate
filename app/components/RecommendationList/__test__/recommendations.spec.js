import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import RecommendationList from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('<RecommendationList />', () => {
  const mockProps = {
    recommendations: [
      {
        id: 1,
        backdrop_path: 'jkfdjfd',
        title: 'title',
      },
    ],
  }
  const component = shallow(<RecommendationList {...mockProps} />)

  it('should return div tag', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})

