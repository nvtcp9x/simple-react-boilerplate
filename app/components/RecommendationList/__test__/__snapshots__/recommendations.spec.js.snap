// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecommendationList /> should return div tag 1`] = `
<styled.div>
  <styled.div
    key="1"
  >
    <styled.img
      alt="title"
      src="https://image.tmdb.org/t/p/w250_and_h141_facejkfdjfd"
    />
    <p>
      title
    </p>
  </styled.div>
</styled.div>
`;
